transcom/mymove
# INT B-20893

## [B-20893](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20893)

## Summary

Fixes the CSV export to use an HQ users selected GBLOC rather than it's default GBLOC when requesting data.

The scenario in which the issue occurs maybe should have been in the AC of either B-20012 or B-19780. Seems to be one of many unforeseen interactions the HQ role had with concurrent work (but I should have caught it since I did both).

Also, adds a tabbing/508 fix.

Also Also, It was also hanging when you tried to export nothing so I disabled it if there is nothing to export and updated from <Link> to <Button> because there is no good way to disable an <a>.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the office app.
2. Login as an HQ user.
3. Export your queue, verify it looks right.
4. Switch to a different GBLOC, export that queue, verify it looks right, repeat for a couple other GBLOCs.
5. If the queue is empty verify the export button is disabled.
6. Repeat for other queues.
7. Logout and log in as a non-HQ office user with a queue, verify their export looks right.
8. Repeat step 7 for a different office user in a different GBLOC.

#### ANDI/508 Fix
1. From your office user's queue page tab through all the selectable elements on the screen.
2. Verify you are able to export the queue with just tabs and enter.
3. You may notice if you tab past that you get stuck in a keyboard trap (stuck on the same input no matter how many times you tab forward) when trying to tab through all of the filter fields. I think this is a 508 Compliance issue that needs to be addressed but I couldn't easily find a fix for it. It seems like there is an event listener hijacking the default behavior.

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
